packages/aws-cdk/lib/api/bootstrap/bootstrap-environment.ts

```diff
@@ -145,12 +166,90 @@ export class Bootstrapper {
         CloudFormationExecutionPolicies: cloudFormationExecutionPolicies.join(','),
         Qualifier: params.qualifier,
         PublicAccessBlockConfiguration: params.publicAccessBlockConfiguration || params.publicAccessBlockConfiguration === undefined ? 'true' : 'false',
+        PermissionsBoundary: permissionsBoundary,
       }, {
         ...options,
         terminationProtection: options.terminationProtection ?? current.terminationProtection,
       });
   }
 
+  private validatePolicyName(permissionsBoundary: string) {
+    // https://docs.aws.amazon.com/IAM/latest/APIReference/API_CreatePolicy.html
+    const regexp: RegExp = /[\w+=,.@-]+/;
+    const matches = regexp.exec(permissionsBoundary);
+    if (!(matches && matches.length === 1 && matches[0] === permissionsBoundary)) {
+      throw new Error('Please pass the custom permissions boundary by name when using \'--permissions-boundary\'');
+    }
+  }
+
+  private createExamplePermissionsBoundary(account: string, region: string, qualifier: string|undefined): string {
+    qualifier = qualifier ? qualifier : 'hnb659fds'; // if none is supplied, resort to the default qualifier
+    const policyDoc = {
+      Version: '2012-10-17',
+      Statement: [
+        {
+          Action: ['*'],
+          Resource: '*',
+          Effect: 'Allow',
+          Sid: 'ExplicitAllowAll',
+        },
+        {
+          Condition: {
+            StringEquals: {
+              'iam:PermissionsBoundary': 'arn:aws:iam::' + account + ':policy/cdk-' + qualifier + '-BootstrapPermissionBoundary-' + account + '-' + region,
+            },
+          },
+          Action: [
+            'iam:CreateUser',
+            'iam:CreateRole',
+            'iam:PutRolePermissionsBoundary',
+            'iam:PutUserPermissionsBoundary',
+          ],
+          Resource: '*',
+          Effect: 'Allow',
+          Sid: 'DenyAccessIfRequiredPermBoundaryIsNotBeingApplied',
+        },
+        {
+          Action: [
+            'iam:CreatePolicyVersion',
+            'iam:DeletePolicy',
+            'iam:DeletePolicyVersion',
+            'iam:SetDefaultPolicyVersion',
+          ],
+          Resource: 'arn:aws:iam::885876182003:policy/cdk-'+qualifier+'-BootstrapPermissionBoundary-' + account + '-' + region,
+          Effect: 'Deny',
+          Sid: 'DenyPermBoundaryIAMPolicyAlteration',
+        },
+        {
+          Action: [
+            'iam:DeleteUserPermissionsBoundary',
+            'iam:DeleteRolePermissionsBoundary',
+          ],
+          Resource: '*',
+          Effect: 'Deny',
+          Sid: 'DenyRemovalOfPermBoundaryFromAnyUserOrRole',
+        },
+      ],
+    };
+    const request = {
+      PolicyName: CDK_BOOTSTRAP_PERMISSIONS_BOUNDARY,
```

Comment:
```suggestion
      PolicyName: `cdk-${qualifier}-permissions-boundary`,
```

Since we are creating this policy from the CLI how about we use a different name
from what is in the template. We also don't need to use the account/region in
the name since we can first lookup and see if the policy exists.